nsv13/code/modules/power/gravitygenerator_modular.dm

```diff
@@ -0,0 +1,73 @@
+///This is the modular NSV file for gravity generators, to keep the base file more clear of conflicts.
+
+//Modular proc attachment
+/obj/machinery/gravity_generator/main/attackby(obj/item/I, mob/user, params)
+	if(!(obj_flags & EMAGGED))
+		return ..()
+	if(I?.tool_behaviour != TOOL_MULTITOOL)
+		return ..()
+	if(on || charge_count > 0 || charging_state == POWER_UP)
+		to_chat(user, "<span class='warning'>You should not mess with this device's circuitry while it is active!</span>")
+		return TRUE
+	if(machine_stat & BROKEN)
+		to_chat(user, "<span class='notice'>Fix [src]'s chassis first.</span>")
+		return TRUE
+	to_chat(user, "<span class='notice'>You start resetting [src]'s control circuits and safety protocols.</span>")
+	if(!do_after(user, 10 SECONDS, target = src, extra_checks = CALLBACK(src, PROC_REF(can_reset_generator))))
+		return TRUE
+	to_chat(user, "<span class='notice'>You reset [src]'s control circuits to default.</span>")
+	obj_flags &= (~EMAGGED)
+	change_setting(1)
+	ui_update()
+	if(ishuman(user))
+		add_fibers(user) //..Fixing this thing is ALSO close and personal work.
+	investigate_log("was reset while emagged and will now generate 1G", INVESTIGATE_GRAVITY)
+	return TRUE
+
+///Is this gravity generator in need of a reset right now, and is this a valid state for it to be reset in?
+/obj/machinery/gravity_generator/main/proc/can_reset_generator()
+	if(on || charge_count > 0 || charging_state == POWER_UP || !(obj_flags & EMAGGED) || (machine_stat & BROKEN))
+		return FALSE
+	return TRUE
+
+///Can this gravity generator be emagged right now?
+/obj/machinery/gravity_generator/main/proc/can_emag_generator()
+	if((obj_flags & EMAGGED) || !on)
+		return FALSE
+	return TRUE
+
+/obj/machinery/gravity_generator/main/emag_act(mob/user)
+	if(obj_flags & EMAGGED)
+		return FALSE
+	if(!on)
+		to_chat(user, "<span class='notice'>[src] must be active to most efficiently override its controls.</span>")
```

Comment:
```suggestion
		to_chat(user, "<span class='notice'>[src] must be active to override its controls.</span>")
```
If it's about efficiency you could still make them able to do it more slowly when off. Otherwise mentioning it seems a bit useless.

Reply:
Well, the "less efficiently" version is "does nothing because certain vulnerabilities the emag hooks into for real impact are only accessible while active.